### PR TITLE
Chore: Add Types To ButtonPanel Component

### PR DESCRIPTION
**Pull request by keyy123**
Outstanding Issues:

I tried to look at web-map-core's frontend views via ```npm run dev``` but I received 400+ errors from webpack I suppose from the other components and types that don't have any type or type annotations on them at the moment.

### CODE REVIEW

**Review by RubenSmn — COMMENTED** [review]
I don't think we need to add/change functionality, the `isHidden`. Since the `_` is a convention to indicate that it is a private field. The `isHidden` function works as a getter.

**Comment by keyy123** [comment]
Ok, I will undo the changes to the isHidden getter function! 👌 

Does the private isHidden function return a boolean? Based on your answer, I can add a private callable type that returns a boolean right now.

File: src/ButtonPanel.ts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+import { ButtonPanelMethods } from './types'
 
-export default class ButtonPanel {
+export default class ButtonPanel implements ButtonPanelMethods {
   leftButtonId = 'next-left-arrow'
   rightButtonId = 'next-right-arrow'
+  buttonPanel: HTMLDivElement
+  leftArrow: HTMLDivElement
+  rightArrow: HTMLDivElement
+  private _isHidden: Boolean
```
Comment (RubenSmn) [inline src/ButtonPanel.ts:16]: `Boolean` is an object type `boolean` is an primitive type, we should always try to use the primitive type. If you want more in depth information checkout the [everyday types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html) page of the TypeScript Lang webiste

File: src/ButtonPanel.ts [hunk 2]
```diff
@@ -79,8 +86,10 @@ export default class ButtonPanel {
     this.buttonPanel.style.display = 'none'
   }
 
-  clickHandler(e) {
-    const targetId = e.target.closest('.next-button-container').id
+  clickHandler(e: Event) {
```
Comment (RubenSmn) [inline src/ButtonPanel.ts:89]: We should make this more explicit by defining what kind of `Event` it is

File: src/types.ts [hunk 3]
```diff
@@ -4,3 +4,14 @@ export type CoordinatesType = {
...
+
+export type ButtonPanelMethods = {
+  hide: () => void
+  showLeftArrow: () => void
+  showRightArrow: () => void
+  hideLeftArrow: () => void
+  hideRightArrow: () => void
+  show: () => void
+  mount: (e: Element) => void
+  clickHandler: (e: Event) => void
```
Comment (RubenSmn) [inline src/types.ts:16]: Same here with the `Event`

**Review by RubenSmn — CHANGES_REQUESTED** [review]
Looks great :+1:, I left some comments with a few things that should be addressed before this gets merged!

File: src/types.ts (hunk 3 above)
Comment (keyy123) [reply src/types.ts:16]: 👌

**Comment by keyy123** [comment]
I made the changes. I appreciate the mix-up.

File: src/ButtonPanel.ts [hunk 4]
```diff
@@ -44,7 +51,7 @@ export default class ButtonPanel {
     this._isHidden = false
   }
 
-  mount(element) {
+  mount(element: Element): void {
```
Comment (RubenSmn) [inline src/ButtonPanel.ts:54]: The `void` is already defined in the `ButtonPanelMethods` :wink:

**Review by RubenSmn — CHANGES_REQUESTED** [review]

File: src/ButtonPanel.ts (hunk 4 above)
Comment (keyy123) [reply src/ButtonPanel.ts:54]: You're right! I'm on it!

**Pull request merged**